### PR TITLE
[moment-range] Fixed extendMoment function any return

### PR DESCRIPTION
**Pull request by MartynasZilinskas**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Fixed `any` return type in `extendMoment` function.

### CODE REVIEW

**Comment by MartynasZilinskas** [comment]
While fixing `moment-range`, I removed `moment` types exports in this package, because the package itself doesn't have an instance of `moment`.

Another problem is that `react-daterange-pick` types use `moment-range` as if it had `moment` types, which it doesn't. So I had to fix it too in the same pr.

[`moment-range` get started](https://github.com/gf3/moment-range/blob/master/README.md#node--npm).

Can you check if everything is alright @andy-ms?

Also pinging `react-daterange-picker` package author: @uncovertruth.

File: types/react-daterange-picker/index.d.ts [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
+// Type definitions for React Daterange Picker 1.1
 // Project: https://github.com/onefinestay/react-daterange-picker
 // Definitions by: UNCOVER TRUTH Inc. <https://github.com/uncovertruth/>
+//                 MartynasZilinskas <https://github.com/MartynasZilinskas>
 // Definitions: https://github.com/borisyankov/DefinitelyTyped
 // TypeScript Version: 2.1
 
 import * as React from "react";
-import * as moment from "moment-range";
+import * as Moment from "moment";
```
Comment (GiedriusGrabauskas) [inline types/react-daterange-picker/index.d.ts]: First letter lower case (`moment`).

File: types/react-daterange-picker/index.d.ts [hunk 2]
```diff
@@ -30,37 +32,35 @@ interface Props extends React.Props<{}> {
     onHighlightDate?(date: Date): void;
     onHighlightRange?(date: Date): void;
     onSelect?(value: Props): void;
-    onSelectStart?(value: moment.Moment): void;
+    onSelectStart?(value: Moment.Moment & MomentRange.MomentRangeExtends): void;
```
Comment (GiedriusGrabauskas) [inline types/react-daterange-picker/index.d.ts]: Create new type for value

**Review by GiedriusGrabauskas — CHANGES_REQUESTED** [review]

File: types/react-daterange-picker/index.d.ts (hunk 1 above)
Comment (MartynasZilinskas) [inline types/react-daterange-picker/index.d.ts]: Fixed.

File: types/react-daterange-picker/index.d.ts (hunk 2 above)
Comment (MartynasZilinskas) [inline types/react-daterange-picker/index.d.ts]: Fixed.

**Review by GiedriusGrabauskas — APPROVED** [review]

File: types/moment-range/index.d.ts [hunk 3]
```diff
@@ -1,104 +1,62 @@
...
-    function extendMoment(moment: any): any;
+    reverseByRange(interval: DateRange, options?: { exclusive?: boolean, step?: number }): Iterable<Moment>;
+}
 
-    function range(range: string | Date[] | Moment[]): Range;
-    function range(start: Date | Moment, end: Date | Moment): Range;
+export interface MomentRangeExtends {
+    range(range: string | Date[] | Moment[]): DateRange;
+    range(start: Date | Moment, end: Date | Moment): DateRange;
 }
```
Comment (franjuan) [inline types/moment-range/index.d.ts]: It makes sense to include within function here, when extending Moment type.
Thanks!

Comment (franjuan) [inline types/moment-range/index.d.ts]:
```typescript
within(range: DateRange): boolean
```

Comment (MartynasZilinskas) [inline types/moment-range/index.d.ts]: Fixed. [Link to source `moment-range`](https://github.com/gf3/moment-range/blob/master/lib/moment-range.js#L329)

**Pull request merged**